column_buffer.go

```diff
@@ -644,7 +646,21 @@ func (col *repeatedColumnBuffer) Values() ValueReader {
 	return &repeatedPageReader{page: col.Page().(*repeatedPage)}
 }
 
-func rowsHaveBeenReordered(rows []region) bool {
+func optionalRowsHaveBeenReordered(rows []int32) bool {
+	i := int32(0)
+	for _, row := range rows {
```

Comment:
```go
for i := 0; i < len(rows); i++ {
    if i != rows[i]
}
```

?

Reply:
That's not quite the same, we can only increase `i` if we actually see a non-null row.

Reply:
I think this is not super obvious so I added some code comments that should clarify it for future readers.